repos/system_upgrade/el7toel8/actors/nischeck/libraries/nischeck.py

```diff
@@ -0,0 +1,166 @@
+            reporting.Severity(reporting.Severity.INFO)
+        ])
+
+    def hostnames_in_yp_conf(self):
+        """
+        Scan /etc/yp.conf, return if file contains domain name
+        """
+
+        path = "/etc/yp.conf"
+        try:
+            with open(path) as f:
+                lines = f.readlines()
+        except IOError:
+            return False
+
+        # remove comments
+        lines = list(map(lambda x: x.split("#", 1)[0], lines))
+
+        re_server = r"\s*domain\s+\S+\s+server\s+(\S+)\s*"
+        re_ypserver = r"\s*ypserver\s+(\S+)\s*"
+
+        # Find server name specified in form: domain NISDOMAIN server HOSTNAME
+        mtchs = list(filter(lambda x: x is not None,
+                            map(lambda x: re.match(re_server, x), lines)))
+
+        # Find server name specified in form: ypserver HOSTNAME
+        mtchs.extend(list(filter(lambda x: x is not None,
+                                 map(lambda x: re.match(re_ypserver, x), lines))))
+
+        # Get hostnames from matches
+        hostnames = list(map(lambda x: x.group(1), mtchs))
+
+        # Get hostnames that are not IP addresses
+        not_ips = list()
+        for h in hostnames:
+            try:
+                ipaddress.ip_address(h)
+            except ValueError:
+                not_ips.append(h)
+
+        return not_ips
+
+    def scan_nsswitch(self):
+        """
+        Read the contents of the ``/etc/nsswitch.conf`` file.
+        Each non-commented line is split into the service and its sources.  The
+        sources (e.g. 'files sss') are stored as is, as a string.
+        nsswitch.conf is case insensitive.  This means that both the service and
+        its sources are converted to lower case and searches should be done
+        using lower case text.
+        Attributes:
+            data (dict): The service dictionary
+            errors (list): Non-blank lines which don't contain a ':'
+            sources (set): An unordered set of the sources seen in this file
+        Sample content::
+            # Example:
+            #passwd:    db files nisplus nis
+            #shadow:    db files nisplus nis
+            #group:     db files nisplus nis
+            passwd:     files sss
+            shadow:     files sss
+            group:      files sss
+            #initgroups: files
+            #hosts:     db files nisplus nis dns
+            hosts:      files dns myhostname
+        Examples:
+            >>> nss = shared[NSSwitchConf]
+            >>> 'passwd' in nss
+            True
+            >>> 'initgroups' in nss
+            False
+            >>> nss['passwd']
+            'files nss'
+            >>> 'files' in nss['hosts']
+            True
+            >>> nss.errors
+            []
+            >>> nss.sources
+            set(['files', 'dns', 'sss', 'myhostname'])
+        """
+
+        path = "/etc/nsswitch.conf"
+        try:
+            with open(path) as f:
+                content = f.readlines()
```

Comment:
that would be a way more efficient to process the content within this context. Something like:
```python
for line in f:
    if line[0] not in ["#", " "]:
        # do something here
```